src/components/settings/SettingsStripePortal.tsx

```diff
@@ -0,0 +1,79 @@
+import { v4 as uuidv4 } from "uuid";
+import { useState } from "react";
+import { useRouter } from "next/router";
+import useUserStore from "@/stores/user";
+import useToastStore from "@/stores/toast";
+import BaseButton from "../base/BaseButton";
+import LoadingSpinner from "@/components/LoadingSpinner";
+import { CreatePortalSessionPayload } from "@/types";
+import { createPortalSessionService } from "@/services/payments";
+
+export default function SettingsStripePortal() {
+  const router = useRouter();
+  const { profile, getPremiumStatus } = useUserStore();
+  const addToast = useToastStore((state) => state.addToast);
+  const [isLoading, setIsLoading] = useState(false);
+
+  async function handleMySubscription() {
+    if (!profile) return;
+
+    try {
+      setIsLoading(true);
+
+      const payload: CreatePortalSessionPayload = { profileId: profile?.id };
+
+      await createPortalSessionService(payload);
+    } catch (error) {
+      addToast({
+        uniqueId: uuidv4(),
```

Comment:
Move `uuidv4` into useToastStore so we don't need to import 2 things to use the toast.